File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.java
class Solution {
    public int equalSubstring(String s, String t, int maxCost) {
        int N=s.length();
        int[] arr = new int[N+1];
        int temp = 0;
        for(int i=0;i<N;i++){
            temp += Math.abs(s.charAt(i) - t.charAt(i));
            arr[i+1]=temp;
        }
        int i=0,j=1,ans=0;
        while(j<N+1){
            if(arr[j]-arr[i]<=maxCost){
                if(ans<j-i) ans=j-i;
                j++;
            }else{
                i++;
            }
        }
        return ans;
    }
}

File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        q=[0]
        ss=0
        for x,y in zip(s,t):
            ss+=abs(ord(x)-ord(y))
            q.append(ss)
        N = len(q)
        l=0
        ans = 0
        r=1
        while r<N:
            if q[r] - q[l] <=maxCost:
                ans=max(ans,r-l)
                r+=1
            else:
                l+=1
        return ans